src/Agent.Worker/Container/DockerCommandManager.cs

```diff
@@ -308,6 +309,17 @@ public async Task<List<PortMapping>> DockerPort(IExecutionContext context, strin
             return DockerUtil.ParseDockerPort(portMappingLines);
         }
 
+        public async Task<bool> IsContainerRunning(IExecutionContext context, string containerId) {
+            List<string> filteredItems = await DockerPS(context, $"--filter id={containerId}");
+
+            if (filteredItems.Count < 2)
+            {
+                return false;
+            }
+
+            return true;
```

Comment:
I think in such way will be better, since there will be less value inversions:

`bool isContainerStopped = (filteredItems.Count < 2);` =>  `const isContainerRunning = (filteredItems.Count == 2);`